### DISCUSSION TITLE
RS485 Wiring question

### DISCUSSION THREAD

**Discussion by sjames99** [aqualinkd/AqualinkD#253] [Q&A] [answered]
My Aqualink has two red 4 pin connectors.  The first one has two devices in 2 and 3 and the other have 3 devices in 2 and 3.  So I tried adding the FTDI wires(Orange and Yellow) to 2 and 3 respectively of the first block, but when I bring the aqualink back online, I lost part of the Aqualink functionality.  Am I putting them in the right place?

![thumbnail_image0 (1)](https://github.com/sfeakes/AqualinkD/assets/145196461/1165cd36-9379-4a34-a46d-4a6bd2faffb7)

**Comment by sjames99** [comment, marked as answer]
Found it, apparently the first set of connectors can only have the board and wifi module.  Placed in the second set of connectors and aqualink was back and running and I could see the serial port transmitting data.

**Comment by sfeakes** [comment]
You should look at the wiki about wiring and now termination works.  It sounds like you may have 4 or 5 devices all connected to the terminal block rather than a daisy chain / in series.   This *might* cause you some issues, so please keep that in mind if things don't work as expected.

**Reply by sjames99** [reply]
Thanks, I haven't had a chance to really dive in just yet.  The only thing I noticed was the SWG was showing 0% and 0 PPM but not sure if it is mapping problem.  I will look into daisy chaining the wires.

**Comment by sjames99** [comment]
Got a little further, haven't really dove in too deep yet due to the time.  I have a question about the serial output.  In the wiki it says you should some words.  I only see this:

```
 sudo -s eval 'stty raw -echo < /dev/ttyUSB0; cat -vte /dev/ttyUSB0'
^B^B^P^B^B^P^B^B^P^B^B^P^B^B^P^B^C^P^B^C^P^B^C^P^B^C^P^B^C^P^B^AM-BJ^P%1^@^C05M-:^P^B^AM-BJ^P%1^@^C0A m>^P%iTp^P^B^Ere^@^C^P%iTp^P^B^Ere^@^C084M-:^@^P^C^B0^@8M-BM-:O^P^P^B%^@4M-B^@O^C^P0%84M-:^@^P^C^B0^@8M-BM-:O^P^P^B-Auain TcM-(^P0qLkoh^P^BAanTcM-(^P-ui u^@^C0qLkoh^P^B^@M-BO^P08M-:^P^B^@M-BO^P%4^@^C08M-:^P^B^I^E^P^B^I^E^P^B^I^E^P^B^I^E^P^B^I^E^P^B^I^E^P^B0^X^WM-P^P^B0^X)^C^B^I^E^P^B^I^E^P^B^@^@l^P$
^P^@^P^B^@^C^Bu^@^B^@^P^B^H^@^B?^P^B^@^Y^P^_(M-@^PA^C^@^@^Cp^@M-^W^P^M^@^P^BM-M^P^AU^P0^C^B^@^@l^P^B^@^@^C`^P^Bu^@^P^A^@^P^PS^C^P^AR^P^L^@^P^BHg^P^@^C0^P^B^@^@l^P$
^P^@^P^B=^P0^C^P^A^S^PS^C^P^AR^P^L^@^P^BHg^P0^CH^P^@^P^B^@^@.^P^@^C3^P^@^B^@^P^B^H^@^B^A^@^Cp^@M-^W^C^@^@g^P^Br^P^B^@^@^C$
^P^@^P^BC^P0^C^P^A^@^CM-#^P^@^@^@^Cp^@^@^C^@^@^P^Br^P^B^@^@^C^B^@^@.^P^@^C3^P^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^@^@^P^Br^P^@^CH^P^@^P^B^@^@.^P2^P^B0^C^P^A^S^PS^C^P^AR^P^L^@^P^BHg^P0^Cx^E^C^@^@^B^Cx^P^BC^@^P^B^@M-S^P^@^@10@^P^B^@^@^C$
^P^@^P^Br^P0^C^P^A^S^PS^C^P^AR^P^L^@^P^BHg^P0^CH^P^@^P^B^@^@.^P^@^C^Bu^@^B^@^P^B^H^@^B?^P^B^I^@^C^@^@^P^Br^P^B^@^@^C$
^P^@^P^Br^P3^P^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^M^@^C0^P^B^@^@^@^C9^P^BM-^S^P3^P^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^M^@^C$
^P^@^P^Br^P3^P^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^B^@^MH^@^@g^P^B^@^@^C:^P^Bu^@^P^A^S^PS^C^P^AR^P^L^@^P^BHg^P^B^@^@^C`^P^Bu^@^P^A^S^PS^C^P^AR^P^L^@^P^BHg^P^B^@^@^C;^P^Bu^@^P^A^S^PS^C^P^AR^P^L^@^P^BHg^P^B^@^@^C`^P^Bu^@^B^@^P^B^H^@^B?^P^B^I^@^C^@^@^P^B^@^@.^PH^P^Bu^@^B^@^P^B^H^@^B?^P^B^L^@^P^BHg^PD#^C^@^@^@^Cx^P^BA^P^B^@^@.^P^@^CM-^@^P^PB^P^BB^P^Bu^@^P^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^@H^@^P^B^B^P^@.^CP^P^P0^C^P^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^@^@^P^B^@^@.^P^@^C^Bu^@^B^@^P^B^H^@^B?^P^B^I^@^C^@^@^P^B^@^@.^P^@^C3^P^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^M^@g^C$
^P^@^P^Br^P0^C^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^@^@g^C$
^@^@.^P^@^C3^P^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^@^@^P^B^@^@.^P^@^C3^P^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^M^@^C$
^P^@^P^Be^P^@^C3^P^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^@^@^P^B^@^@.^P^@^C3^P^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^@^@^P^B^@^@^@^P^Bj^P0^C^P^A^S^PS^C^P^B?^@^C^P^L^I^@M-^W^P^B^@^M^@g^P^B^@^P^@^@^C^P^@r^P^Bu^P^@^P^A^@^P^CM-#S^C^@^B^@^@R^P^B^I^@M-^W^P^B^@^@^@^C^P^B^@^@^@^P^Ch^@^C^P0u^@^@^@^A^S^P^BM-#^P^C^P^B?^@^C^P^L^I^@M-^W^P^BH^@^P^C$
^B^@^@.^P^B`^P^C30^C^@^B^@^@^S^P^BM-#S^H^P^C^@^B?^P^B^I^@^C^@^M^@g^P^B^@^@^C^B{^P3^P^@^@^@^CM-#^P^@^@^@^Cp^@M-^W^P^M^@^C$
^P^@^P^C

```
Is this good, or an indicator of a bad connection?

**Comment by Bugman1400** [comment]
That doesn't look right. You should be able to identify some readable words like SPAMODE etc that I believe are just plain ASCII.

UPDATE: Checked again and I get similar output from the RPi, but I did notice that the Red TX light on my RS485toUSB converter was not blinking indicating that the comm was halted while the ttyUSB0 was being logged. So, perhaps the CPU on the RPi is maxxed out easily. Coulda swore I saw some readable ASCII when I initially set it up. Think its safe to say that if both the Green and Red LEDs are blinking on the RS485toUSB converter, then the connection is good.

**Comment by ballle98** [comment]
@sjames99 you may want to also hook up ground if your FTDI converter supports it.  It may help with power surges and storms.
